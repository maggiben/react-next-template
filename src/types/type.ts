

export type Person = {
  id: string;
  name: string;
  age: number,
  identification: {
    type: string,
    number: number;
  },
  profession: string;
  faceapi: boolean;
  email: {
    address: string;
    confirmed: boolean;
  }
  city: string;
  cp: string;  
  selected: boolean;
  status: {
    label: string;
    color: string;
  }
}
