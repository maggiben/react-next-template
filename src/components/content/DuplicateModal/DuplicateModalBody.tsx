import * as utils from '@utils/index';
import { 
  TableView, 
  Column, 
  Cell, 
  Label, 
  Radio, 
  CheckCircleIcon, 
  CloseCircleIcon
} from '@fravega-it/bumeran-ds-fvg'
import { useTranslation } from 'react-i18next';
import { Person } from 'types/type';


type DuplicateModalBodyProps = {
  persons: Person[];
  handleSelection: ((checked: boolean, meta: { event: React.ChangeEvent<HTMLInputElement>; value: string; }) => void) | undefined;
}

const DuplicateModalBody = (props: DuplicateModalBodyProps) => {
  const { t } = useTranslation();
  const { persons, handleSelection } = props;

  return (
    <TableView
			items={persons}
			renderColumns={() => (
				<>
					<Column minWidth={20} label={t('select')} />
					<Column minWidth={70} label={t('email')} />
				</>
			)}
			renderCells={({ id, name, selected, email, lastname }) => (
				<>
					<Cell><Radio id={id} name={name} label={name + ' ' + lastname} value={id} checked={selected} onChange={handleSelection}/></Cell>
					<Cell>{email.address}</Cell>
				</>
			)}
    />
  );
};

  export default DuplicateModalBody;