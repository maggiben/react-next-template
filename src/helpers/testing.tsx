import { default as baseRenderer } from 'react-test-renderer';
import { render, RenderResult } from '@testing-library/react';
import i18n from '@utils/i18n';
import { I18nextProvider } from 'react-i18next';
import { RecoilRoot } from 'recoil';
import { ThemeProvider } from 'styled-components';
import { theme, IDefaultTheme } from '@fravega-it/bumeran-ds-fvg';
import React, { ReactElement } from 'react';
import '@testing-library/jest-dom'

/**
 * Helper used to render components for testing.
 * The function appends the theme to the children.
 * */
interface IRenderer {
  create: (children: ReactElement) => RenderResult;
  baseRenderer: (children: ReactElement) => baseRenderer.ReactTestRenderer;
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme) => RenderResult;
  renderWithI18n: (children: ReactElement) => RenderResult;
}

export const renderer: IRenderer = {
  create: (children: ReactElement): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  baseRenderer: (children: ReactElement): baseRenderer.ReactTestRenderer =>
    baseRenderer.create(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithI18n: (children: ReactElement): RenderResult => 
    render(
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
      </ThemeProvider>
  ),
};