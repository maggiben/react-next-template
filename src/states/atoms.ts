import { atom } from 'recoil';

export const personState = atom({
  key: 'personState',
  default: {
    id: '',
    name: '',
    lastname: '',
    age: 0,
    identification: {
      type: '',
      number: 0,
      expiration: '',
    },
    faceapi: false,
    email: {
      address: '',
      confirmed: false,
    },
    renaper: {
      confirmed: false,
      expiration: '',
    },
    profession: '',
    city: '',
    cp: '',
    selected: false,
    status: {
      label: '',
      color: '',
    },
  },
});

export const currentFilterState = atom({
  key: 'currentFilterState',
  default: {
    name: undefined, 
    email: undefined, 
    cuid: undefined,
    documentType: undefined,
    documentNumber: undefined,
  },
});
