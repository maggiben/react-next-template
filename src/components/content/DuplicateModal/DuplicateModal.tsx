import { useEffect, useState, useCallback } from 'react';
import { useRecoilState } from 'recoil';
import { Modal, ModalHeader, ModalBody, ModalFooter, NotificationIcon, TextBody, ButtonGroup, Button } from '@fravega-it/bumeran-ds-fvg';
import { TableView, Column, Cell, Label, Radio } from '@fravega-it/bumeran-ds-fvg'
import { Person } from '../index';
import { personState } from '../../../states/atoms';

type DuplicateModalProps = {
  isOpen?: boolean;
  closeModal: (person?: Person) => void;
  onSelectPersonModal: (person?: Person) => void;
  persons: Person[];
}

const DuplicateModal = (props: DuplicateModalProps) => {
  const [person, setPerson] = useRecoilState(personState);
  const {isOpen, onSelectPersonModal, closeModal, persons} = props;
  const [personsObj, setPersonsObj] = useState<Person[]>({...props.persons});

  useEffect(() => {
    setPersonsObj(props.persons);
  }, [props.persons])

  const updatePersonSelectedStatus = useCallback(
    (id: string) => {
      const p = personsObj.map((person) => {
        if(person.id === id) {
          person.selected = true;
        } else {
          person.selected = false;
        }
        return person;
      });
      setPersonsObj(p)
    },
    [personsObj]
  );

  const setSelectedPerson = () => {
    setPerson(personsObj[personsObj.findIndex((person) => person.selected)]);
  };

  const handleChange = (id: string, selected: boolean) => () => {
    updatePersonSelectedStatus(id);
  };

  return (
    <Modal onClose={closeModal} open={isOpen}>
      <ModalHeader
        leftIcon={<NotificationIcon />}
        title="Duplicado"
        description="Cliente duplicado"
      />

      <ModalBody>
        <TableView
          items={personsObj}
          renderColumns={() => (
            <>
              <Column minWidth={20} label="select" />
              <Column minWidth={50} label="name" />
              <Column minWidth={100} label="faceapi" />
              <Column minWidth={70} label="email" />
              <Column minWidth={70} label="email confirmed" />
              <Column minWidth={100} label="city" />
            </>
          )}
          renderCells={({ id, name, selected, faceapi, email, city }) => (
            <>
              {/* <Cell><Radio id={name} label="Radio" onChange={handleChange} name={name} value={name} /></Cell> */}
              {/* <Cell><Radio id={name} label={name} value={name} onChange={handleChange}/></Cell> */}


              <Cell><Radio id={id} label={name} value={name} checked={selected}  onChange={handleChange(id, selected)}/></Cell>
              <Cell>{name}</Cell>
              <Cell><Label label={faceapi.toString()} color={faceapi ? "green" : "red"} /></Cell>
              <Cell> {email.address}</Cell>
              <Cell><Label label={email.confirmed.toString()} color={email.confirmed ? "green" : "red"} /></Cell>
              <Cell> {city}</Cell>
            </>
          )}
        />
      </ModalBody>

      <ModalFooter>
        <ButtonGroup
          size="s"
          align="left"
          primaryLabel="Accept"
          onClickPrimary={() => {setSelectedPerson(); onSelectPersonModal(personsObj[personsObj.findIndex((person) => person.selected)]);}}
          secondaryLabel="Cancel"
          onClickSecondary={closeModal}
        />
      </ModalFooter>
    </Modal>
  );
};

  export default DuplicateModal;