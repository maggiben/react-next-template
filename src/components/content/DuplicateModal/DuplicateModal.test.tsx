/* eslint-env jest */
import '@testing-library/jest-dom';
import { expect } from '@jest/globals';
import React from 'react';
import { fireEvent } from '@testing-library/react';
import { renderer } from '@helpers/testing';
import { RecoilRoot } from 'recoil';
import DuplicateModal from './DuplicateModal';
import { personsState } from '@states/atoms';
import { Person } from 'types/type';
import i18next from 'i18next';
global.structuredClone = (val) => JSON.parse(JSON.stringify(val));

describe('DuplicateModal', () => {
  it('renders with default values and handles selection', () => {
    const onSelectPersonModal = jest.fn();
    const closeModal = jest.fn();
    const mockedPerson = {
      id: '123',
      name: 'John',
      lastname: 'Doe',
      age: '30',
      validations: {
        email: true,
        renaper: false,
        facephi: true,
      },
      identification: {
        type: 'ID',
        number: 123456789,
        expiration: '2022-12-31',
      },
      email: {
        address: 'john.doe@example.com',
        confirmed: true,
        expiration: '2022-12-31',
      },
      selected: false,
      status: {
        label: 'Active',
        color: 'green',
      },
      phone: {
        number: 987654321,
        prefix: 1,
        codeArea: 123,
      },
      address: {
        zipCode: '12345',
        street: 'Main Street',
        number: '123',
        floor: '2',
        apartment: 'A',
        city: 'Cityville',
        state: 'CA',
        country: 'USA',
      },
    };
    const mockedTom = {...mockedPerson, id: '543', name: 'Tom'};
    const mockedPersons = [mockedPerson, mockedPerson, mockedTom, mockedPerson];
    
    const tree = renderer.create(
      <RecoilRoot initializeState={(snapshot) => snapshot.set(personsState, mockedPersons)}>
        <DuplicateModal isOpen={true} onSelectPersonModal={onSelectPersonModal} closeModal={closeModal} />
      </RecoilRoot>
    );
    expect(tree.baseElement).toMatchSnapshot();

    const acceptButton = tree.getByRole('button', { name: i18next.t('accept') });//, { name: 'Accept' })[0];
    const selection = tree.getByRole('radio', { name: 'Tom Doe'});
    // console.log('tomRadioButtons', tomRadioButtons);
    // const radioButtons = tree.getAllByRole('radio');
    // const selection = radioButtons[2];
    fireEvent.click(selection);
    fireEvent.click(acceptButton);
    expect(onSelectPersonModal).toHaveBeenCalledWith({...mockedTom, selected: true});
    // expect(closeModal).toHaveBeenCalled();
    // expect(tree.baseElement).toMatchSnapshot();
  });
});

