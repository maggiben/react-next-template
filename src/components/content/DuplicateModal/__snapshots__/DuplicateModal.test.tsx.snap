// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DuplicateModal renders with default values and handles empty selection 1`] = `
<body
  style="overflow: hidden;"
>
  <div>
    <div
      class="sc-gmgFlS cmxTsa"
    >
      <div
        class="sc-kMkxaj fEipSc"
        data-testid="header"
      >
        <svg
          class="sc-kpDqfm gtIuVz"
          color="black"
          fill="none"
          viewBox="0 0 28 28"
          width="28px"
        >
          <path
            clip-rule="evenodd"
            d="M11.681 8.295a2.334 2.334 0 0 1 4.638 0 5.445 5.445 0 0 1 3.126 4.927v2.457c0 .213.084.417.235.567l1.092 1.093a.778.778 0 0 1-.55 1.328h-3.11a3.111 3.111 0 0 1-6.223 0H7.778a.778.778 0 0 1-.55-1.328l1.093-1.093a.802.802 0 0 0 .235-.567v-2.457a5.445 5.445 0 0 1 3.125-4.927Zm.764 10.372a1.556 1.556 0 1 0 3.11 0h-3.11ZM14 7.777a.778.778 0 0 0-.778.779v.265c0 .33-.207.624-.518.733a3.89 3.89 0 0 0-2.593 3.668v2.457c0 .52-.172 1.023-.485 1.432h8.748a2.358 2.358 0 0 1-.485-1.432v-2.457a3.89 3.89 0 0 0-2.593-3.668.778.778 0 0 1-.518-.733v-.265A.778.778 0 0 0 14 7.777Z"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="sc-fiCwlc bfzQg"
        >
          <div
            aria-level="4"
            class="sc-kAyceB kxtWXq"
            role="heading"
          >
            Duplicado
          </div>
          <span
            class="sc-dcJsrY hnmyIZ"
          >
            Cliente duplicado
          </span>
        </div>
        <button
          class="sc-fHjqPf gFiHfi"
        >
          <svg
            class="sc-kpDqfm ldXwSI"
            color="neutral"
            fill="none"
            viewBox="0 0 28 28"
            width="20px"
          >
            <path
              clip-rule="evenodd"
              d="M8.783 9.283a.778.778 0 0 1 1.1 0L14 13.4l4.117-4.117a.778.778 0 1 1 1.1 1.1L15.1 14.5l4.117 4.117a.778.778 0 1 1-1.1 1.1L14 15.6l-4.117 4.117a.778.778 0 1 1-1.1-1.1L12.9 14.5l-4.117-4.117a.778.778 0 0 1 0-1.1Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <div
        class="sc-jIGnZt jiqWkj"
      >
        <div
          class="sc-eZkCL bFcIFk"
        >
          <table
            class="sc-jnOGJG fcvICe"
          >
            <thead
              class="sc-iMWBiJ emBygi"
            >
              <tr>
                <th
                  class="sc-bypJrT iVfYDj"
                >
                  <div
                    class="sc-ddjGPC dzgfUq"
                  >
                    <div
                      aria-level="6"
                      class="sc-kAyceB ffHZfz"
                      color="inherit"
                      role="heading"
                    >
                      Seleccionar
                    </div>
                  </div>
                </th>
                <th
                  class="sc-bypJrT kTPjQW"
                >
                  <div
                    class="sc-ddjGPC dzgfUq"
                  >
                    <div
                      aria-level="6"
                      class="sc-kAyceB ffHZfz"
                      color="inherit"
                      role="heading"
                    >
                      Email
                    </div>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="sc-gFVvzn iqQWZp"
            >
              <tr
                class="sc-brPLxw igJPLm"
              >
                <td
                  class="sc-klVQfs ebmFut"
                >
                  <p
                    class="sc-gsFSXq crxiSA"
                    color="inherit"
                  >
                    <label
                      class="sc-aXZVg TdWnH"
                    >
                      <input
                        aria-checked="false"
                        class="sc-gEvEer gnHvVk"
                        id="123"
                        name="John"
                        type="radio"
                        value="123"
                      />
                      <div
                        class="sc-fqkvVR dEKrYP"
                        type="radio"
                      />
                      <div
                        class="sc-eqUAAy hHIDJi"
                        type="radio"
                      >
                        <span
                          aria-label="John Doe"
                          class="sc-iGgWBj djeqlO"
                          color="inherit"
                        >
                          John Doe
                        </span>
                      </div>
                    </label>
                  </p>
                </td>
                <td
                  class="sc-klVQfs ebmFut"
                >
                  <p
                    class="sc-gsFSXq crxiSA"
                    color="inherit"
                  >
                    john.doe@example.com
                  </p>
                </td>
              </tr>
              <tr
                class="sc-brPLxw igJPLm"
              >
                <td
                  class="sc-klVQfs ebmFut"
                >
                  <p
                    class="sc-gsFSXq crxiSA"
                    color="inherit"
                  >
                    <label
                      class="sc-aXZVg TdWnH"
                    >
                      <input
                        aria-checked="false"
                        class="sc-gEvEer gnHvVk"
                        id="123"
                        name="John"
                        type="radio"
                        value="123"
                      />
                      <div
                        class="sc-fqkvVR dEKrYP"
                        type="radio"
                      />
                      <div
                        class="sc-eqUAAy hHIDJi"
                        type="radio"
                      >
                        <span
                          aria-label="John Doe"
                          class="sc-iGgWBj djeqlO"
                          color="inherit"
                        >
                          John Doe
                        </span>
                      </div>
                    </label>
                  </p>
                </td>
                <td
                  class="sc-klVQfs ebmFut"
                >
                  <p
                    class="sc-gsFSXq crxiSA"
                    color="inherit"
                  >
                    john.doe@example.com
                  </p>
                </td>
              </tr>
              <tr
                class="sc-brPLxw igJPLm"
              >
                <td
                  class="sc-klVQfs ebmFut"
                >
                  <p
                    class="sc-gsFSXq crxiSA"
                    color="inherit"
                  >
                    <label
                      class="sc-aXZVg TdWnH"
                    >
                      <input
                        aria-checked="false"
                        class="sc-gEvEer gnHvVk"
                        id="543"
                        name="Tom"
                        type="radio"
                        value="543"
                      />
                      <div
                        class="sc-fqkvVR dEKrYP"
                        type="radio"
                      />
                      <div
                        class="sc-eqUAAy hHIDJi"
                        type="radio"
                      >
                        <span
                          aria-label="Tom Doe"
                          class="sc-iGgWBj djeqlO"
                          color="inherit"
                        >
                          Tom Doe
                        </span>
                      </div>
                    </label>
                  </p>
                </td>
                <td
                  class="sc-klVQfs ebmFut"
                >
                  <p
                    class="sc-gsFSXq crxiSA"
                    color="inherit"
                  >
                    john.doe@example.com
                  </p>
                </td>
              </tr>
              <tr
                class="sc-brPLxw igJPLm"
              >
                <td
                  class="sc-klVQfs ebmFut"
                >
                  <p
                    class="sc-gsFSXq crxiSA"
                    color="inherit"
                  >
                    <label
                      class="sc-aXZVg TdWnH"
                    >
                      <input
                        aria-checked="false"
                        class="sc-gEvEer gnHvVk"
                        id="123"
                        name="John"
                        type="radio"
                        value="123"
                      />
                      <div
                        class="sc-fqkvVR dEKrYP"
                        type="radio"
                      />
                      <div
                        class="sc-eqUAAy hHIDJi"
                        type="radio"
                      >
                        <span
                          aria-label="John Doe"
                          class="sc-iGgWBj djeqlO"
                          color="inherit"
                        >
                          John Doe
                        </span>
                      </div>
                    </label>
                  </p>
                </td>
                <td
                  class="sc-klVQfs ebmFut"
                >
                  <p
                    class="sc-gsFSXq crxiSA"
                    color="inherit"
                  >
                    john.doe@example.com
                  </p>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="sc-fBdRDi esHGLi"
      >
        <div
          class="sc-bXCLTC iOgQpC"
        >
          <button
            class="sc-fHjqPf fZZbpa"
          >
            <label
              aria-label="Aceptar"
            >
              <div
                aria-level="6"
                class="sc-kAyceB ffHZfz"
                color="inherit"
                role="heading"
              >
                Aceptar
              </div>
            </label>
          </button>
          <button
            class="sc-fHjqPf jVpOmm"
          >
            <label
              aria-label="Cancelar"
            >
              <div
                aria-level="6"
                class="sc-kAyceB ffHZfz"
                color="inherit"
                role="heading"
              >
                Cancelar
              </div>
            </label>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`DuplicateModal renders with default values and handles selection 1`] = `
<body
  style="overflow: hidden;"
>
  <div>
    <div
      class="sc-gmgFlS cmxTsa"
    >
      <div
        class="sc-kMkxaj fEipSc"
        data-testid="header"
      >
        <svg
          class="sc-kpDqfm gtIuVz"
          color="black"
          fill="none"
          viewBox="0 0 28 28"
          width="28px"
        >
          <path
            clip-rule="evenodd"
            d="M11.681 8.295a2.334 2.334 0 0 1 4.638 0 5.445 5.445 0 0 1 3.126 4.927v2.457c0 .213.084.417.235.567l1.092 1.093a.778.778 0 0 1-.55 1.328h-3.11a3.111 3.111 0 0 1-6.223 0H7.778a.778.778 0 0 1-.55-1.328l1.093-1.093a.802.802 0 0 0 .235-.567v-2.457a5.445 5.445 0 0 1 3.125-4.927Zm.764 10.372a1.556 1.556 0 1 0 3.11 0h-3.11ZM14 7.777a.778.778 0 0 0-.778.779v.265c0 .33-.207.624-.518.733a3.89 3.89 0 0 0-2.593 3.668v2.457c0 .52-.172 1.023-.485 1.432h8.748a2.358 2.358 0 0 1-.485-1.432v-2.457a3.89 3.89 0 0 0-2.593-3.668.778.778 0 0 1-.518-.733v-.265A.778.778 0 0 0 14 7.777Z"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="sc-fiCwlc bfzQg"
        >
          <div
            aria-level="4"
            class="sc-kAyceB kxtWXq"
            role="heading"
          >
            Duplicado
          </div>
          <span
            class="sc-dcJsrY hnmyIZ"
          >
            Cliente duplicado
          </span>
        </div>
        <button
          class="sc-fHjqPf gFiHfi"
        >
          <svg
            class="sc-kpDqfm ldXwSI"
            color="neutral"
            fill="none"
            viewBox="0 0 28 28"
            width="20px"
          >
            <path
              clip-rule="evenodd"
              d="M8.783 9.283a.778.778 0 0 1 1.1 0L14 13.4l4.117-4.117a.778.778 0 1 1 1.1 1.1L15.1 14.5l4.117 4.117a.778.778 0 1 1-1.1 1.1L14 15.6l-4.117 4.117a.778.778 0 1 1-1.1-1.1L12.9 14.5l-4.117-4.117a.778.778 0 0 1 0-1.1Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <div
        class="sc-jIGnZt jiqWkj"
      >
        <div
          class="sc-eZkCL bFcIFk"
        >
          <table
            class="sc-jnOGJG fcvICe"
          >
            <thead
              class="sc-iMWBiJ emBygi"
            >
              <tr>
                <th
                  class="sc-bypJrT iVfYDj"
                >
                  <div
                    class="sc-ddjGPC dzgfUq"
                  >
                    <div
                      aria-level="6"
                      class="sc-kAyceB ffHZfz"
                      color="inherit"
                      role="heading"
                    >
                      Seleccionar
                    </div>
                  </div>
                </th>
                <th
                  class="sc-bypJrT kTPjQW"
                >
                  <div
                    class="sc-ddjGPC dzgfUq"
                  >
                    <div
                      aria-level="6"
                      class="sc-kAyceB ffHZfz"
                      color="inherit"
                      role="heading"
                    >
                      Email
                    </div>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="sc-gFVvzn iqQWZp"
            >
              <tr
                class="sc-brPLxw igJPLm"
              >
                <td
                  class="sc-klVQfs ebmFut"
                >
                  <p
                    class="sc-gsFSXq crxiSA"
                    color="inherit"
                  >
                    <label
                      class="sc-aXZVg TdWnH"
                    >
                      <input
                        aria-checked="false"
                        class="sc-gEvEer gnHvVk"
                        id="123"
                        name="John"
                        type="radio"
                        value="123"
                      />
                      <div
                        class="sc-fqkvVR dEKrYP"
                        type="radio"
                      />
                      <div
                        class="sc-eqUAAy hHIDJi"
                        type="radio"
                      >
                        <span
                          aria-label="John Doe"
                          class="sc-iGgWBj djeqlO"
                          color="inherit"
                        >
                          John Doe
                        </span>
                      </div>
                    </label>
                  </p>
                </td>
                <td
                  class="sc-klVQfs ebmFut"
                >
                  <p
                    class="sc-gsFSXq crxiSA"
                    color="inherit"
                  >
                    john.doe@example.com
                  </p>
                </td>
              </tr>
              <tr
                class="sc-brPLxw igJPLm"
              >
                <td
                  class="sc-klVQfs ebmFut"
                >
                  <p
                    class="sc-gsFSXq crxiSA"
                    color="inherit"
                  >
                    <label
                      class="sc-aXZVg TdWnH"
                    >
                      <input
                        aria-checked="false"
                        class="sc-gEvEer gnHvVk"
                        id="123"
                        name="John"
                        type="radio"
                        value="123"
                      />
                      <div
                        class="sc-fqkvVR dEKrYP"
                        type="radio"
                      />
                      <div
                        class="sc-eqUAAy hHIDJi"
                        type="radio"
                      >
                        <span
                          aria-label="John Doe"
                          class="sc-iGgWBj djeqlO"
                          color="inherit"
                        >
                          John Doe
                        </span>
                      </div>
                    </label>
                  </p>
                </td>
                <td
                  class="sc-klVQfs ebmFut"
                >
                  <p
                    class="sc-gsFSXq crxiSA"
                    color="inherit"
                  >
                    john.doe@example.com
                  </p>
                </td>
              </tr>
              <tr
                class="sc-brPLxw igJPLm"
              >
                <td
                  class="sc-klVQfs ebmFut"
                >
                  <p
                    class="sc-gsFSXq crxiSA"
                    color="inherit"
                  >
                    <label
                      class="sc-aXZVg TdWnH"
                    >
                      <input
                        aria-checked="false"
                        class="sc-gEvEer gnHvVk"
                        id="543"
                        name="Tom"
                        type="radio"
                        value="543"
                      />
                      <div
                        class="sc-fqkvVR dEKrYP"
                        type="radio"
                      />
                      <div
                        class="sc-eqUAAy hHIDJi"
                        type="radio"
                      >
                        <span
                          aria-label="Tom Doe"
                          class="sc-iGgWBj djeqlO"
                          color="inherit"
                        >
                          Tom Doe
                        </span>
                      </div>
                    </label>
                  </p>
                </td>
                <td
                  class="sc-klVQfs ebmFut"
                >
                  <p
                    class="sc-gsFSXq crxiSA"
                    color="inherit"
                  >
                    john.doe@example.com
                  </p>
                </td>
              </tr>
              <tr
                class="sc-brPLxw igJPLm"
              >
                <td
                  class="sc-klVQfs ebmFut"
                >
                  <p
                    class="sc-gsFSXq crxiSA"
                    color="inherit"
                  >
                    <label
                      class="sc-aXZVg TdWnH"
                    >
                      <input
                        aria-checked="false"
                        class="sc-gEvEer gnHvVk"
                        id="123"
                        name="John"
                        type="radio"
                        value="123"
                      />
                      <div
                        class="sc-fqkvVR dEKrYP"
                        type="radio"
                      />
                      <div
                        class="sc-eqUAAy hHIDJi"
                        type="radio"
                      >
                        <span
                          aria-label="John Doe"
                          class="sc-iGgWBj djeqlO"
                          color="inherit"
                        >
                          John Doe
                        </span>
                      </div>
                    </label>
                  </p>
                </td>
                <td
                  class="sc-klVQfs ebmFut"
                >
                  <p
                    class="sc-gsFSXq crxiSA"
                    color="inherit"
                  >
                    john.doe@example.com
                  </p>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="sc-fBdRDi esHGLi"
      >
        <div
          class="sc-bXCLTC iOgQpC"
        >
          <button
            class="sc-fHjqPf fZZbpa"
          >
            <label
              aria-label="Aceptar"
            >
              <div
                aria-level="6"
                class="sc-kAyceB ffHZfz"
                color="inherit"
                role="heading"
              >
                Aceptar
              </div>
            </label>
          </button>
          <button
            class="sc-fHjqPf jVpOmm"
          >
            <label
              aria-label="Cancelar"
            >
              <div
                aria-level="6"
                class="sc-kAyceB ffHZfz"
                color="inherit"
                role="heading"
              >
                Cancelar
              </div>
            </label>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
